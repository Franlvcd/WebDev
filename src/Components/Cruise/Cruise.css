.cruise {
  display: flex;
  text-align: center;
  vertical-align: center;
  padding: 50px 0;
  justify-content: center;
}

.footer-row {
  margin-bottom: 7px;
  font-weight: bold;
  font-size: 18px;
}

.footer-content-row {
  margin-bottom: 3px;
}
