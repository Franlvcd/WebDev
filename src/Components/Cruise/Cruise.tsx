import { Row, Col, Divider } from 'antd'
import './Cruise.css'
import FooterColumn from './FooterColumn/FooterColumn'

const footerStyle = {
  marginBottom: '20px',
  display: 'flex',
  justifyContent: 'center',
}

export const Cruise = () => {
  return (
    <div className="app-footer" data-testid="footer-elements">
      <Row gutter={16} justify="center" align="top">
        <Col xs={24} sm={12} md={6} lg={5} style={footerStyle}>
          <FooterColumn
            footerHeader={'Discover'}
            footerElements={[
              'Products',
              'Trails',
              'Services',
              'Software',
              'Industries',
              'Demos',
            ]}
            spanColumn={6}
          ></FooterColumn>
        </Col>
        <Col xs={24} sm={12} md={6} lg={5} style={footerStyle}>
          <FooterColumn
            footerHeader={'Information'}
            footerElements={['Developers', 'Marketers', 'Education', 'Status']}
            spanColumn={6}
          ></FooterColumn>
        </Col>
        <Col xs={24} sm={12} md={6} lg={5} style={footerStyle}>
          <FooterColumn
            footerHeader={'Contact Us'}
            footerElements={['Support', 'Find a solution']}
            spanColumn={6}
          ></FooterColumn>
        </Col>

        <Col xs={24} sm={12} md={6} lg={5} style={footerStyle}>
          <FooterColumn
            footerHeader={'About'}
            footerElements={[
              'About us',
              'Careers',
              'Latest news',
              'Partnership',
            ]}
            spanColumn={6}
          ></FooterColumn>
        </Col>
      </Row>
      <Divider />

      <Row justify="center">Copyright 2021. Designed by Frances Yang.</Row>
    </div>
  )
}
