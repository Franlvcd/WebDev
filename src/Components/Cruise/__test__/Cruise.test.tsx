import ReactDOM from 'react-dom'
import { Cruise } from '../Cruise'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import renderer from 'react-test-renderer'

global.matchMedia =
  global.matchMedia ||
  function () {
    return {
      addListener: jest.fn(),
      removeListener: jest.fn(),
    }
  }

it('renders without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(<Cruise />, div)
})

it('renders footer columns correctly', () => {
  const { getByTestId } = render(<Cruise />)
  expect(getByTestId('footer-elements')).toHaveTextContent(
    /(?=.*Discover)(?=.*Information)(?=.*Contact Us)(?=.*About)(?=.*Software)(?=.*Partnership)/
  )
})

it('matches snapshot', () => {
  const tree = renderer.create(<Cruise />).toJSON()
  expect(tree).toMatchSnapshot()
})
