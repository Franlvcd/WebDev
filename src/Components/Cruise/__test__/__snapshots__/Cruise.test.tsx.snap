// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="app-footer"
  data-testid="footer-elements"
>
  <div
    className="ant-row ant-row-center ant-row-top"
    style={
      Object {
        "marginLeft": -8,
        "marginRight": -8,
      }
    }
  >
    <div
      className="ant-col ant-col-xs-24 ant-col-sm-12 ant-col-md-6 ant-col-lg-5"
      style={
        Object {
          "display": "flex",
          "justifyContent": "center",
          "marginBottom": "20px",
          "paddingLeft": 8,
          "paddingRight": 8,
        }
      }
    >
      <div
        className="footer-element"
      >
        <div
          className="ant-row footer-row"
          style={Object {}}
        >
          Discover
        </div>
        <div>
          <div
            className="ant-row"
            style={
              Object {
                "fontSize": "20px",
                "marginBottom": "7px",
              }
            }
          >
            Products
          </div>
          <div
            className="ant-row"
            style={
              Object {
                "fontSize": "20px",
                "marginBottom": "7px",
              }
            }
          >
            Trails
          </div>
          <div
            className="ant-row"
            style={
              Object {
                "fontSize": "20px",
                "marginBottom": "7px",
              }
            }
          >
            Services
          </div>
          <div
            className="ant-row"
            style={
              Object {
                "fontSize": "20px",
                "marginBottom": "7px",
              }
            }
          >
            Software
          </div>
          <div
            className="ant-row"
            style={
              Object {
                "fontSize": "20px",
                "marginBottom": "7px",
              }
            }
          >
            Industries
          </div>
          <div
            className="ant-row"
            style={
              Object {
                "fontSize": "20px",
                "marginBottom": "7px",
              }
            }
          >
            Demos
          </div>
        </div>
      </div>
    </div>
    <div
      className="ant-col ant-col-xs-24 ant-col-sm-12 ant-col-md-6 ant-col-lg-5"
      style={
        Object {
          "display": "flex",
          "justifyContent": "center",
          "marginBottom": "20px",
          "paddingLeft": 8,
          "paddingRight": 8,
        }
      }
    >
      <div
        className="footer-element"
      >
        <div
          className="ant-row footer-row"
          style={Object {}}
        >
          Information
        </div>
        <div>
          <div
            className="ant-row"
            style={
              Object {
                "fontSize": "20px",
                "marginBottom": "7px",
              }
            }
          >
            Developers
          </div>
          <div
            className="ant-row"
            style={
              Object {
                "fontSize": "20px",
                "marginBottom": "7px",
              }
            }
          >
            Marketers
          </div>
          <div
            className="ant-row"
            style={
              Object {
                "fontSize": "20px",
                "marginBottom": "7px",
              }
            }
          >
            Education
          </div>
          <div
            className="ant-row"
            style={
              Object {
                "fontSize": "20px",
                "marginBottom": "7px",
              }
            }
          >
            Status
          </div>
        </div>
      </div>
    </div>
    <div
      className="ant-col ant-col-xs-24 ant-col-sm-12 ant-col-md-6 ant-col-lg-5"
      style={
        Object {
          "display": "flex",
          "justifyContent": "center",
          "marginBottom": "20px",
          "paddingLeft": 8,
          "paddingRight": 8,
        }
      }
    >
      <div
        className="footer-element"
      >
        <div
          className="ant-row footer-row"
          style={Object {}}
        >
          Contact Us
        </div>
        <div>
          <div
            className="ant-row"
            style={
              Object {
                "fontSize": "20px",
                "marginBottom": "7px",
              }
            }
          >
            Support
          </div>
          <div
            className="ant-row"
            style={
              Object {
                "fontSize": "20px",
                "marginBottom": "7px",
              }
            }
          >
            Find a solution
          </div>
        </div>
      </div>
    </div>
    <div
      className="ant-col ant-col-xs-24 ant-col-sm-12 ant-col-md-6 ant-col-lg-5"
      style={
        Object {
          "display": "flex",
          "justifyContent": "center",
          "marginBottom": "20px",
          "paddingLeft": 8,
          "paddingRight": 8,
        }
      }
    >
      <div
        className="footer-element"
      >
        <div
          className="ant-row footer-row"
          style={Object {}}
        >
          About
        </div>
        <div>
          <div
            className="ant-row"
            style={
              Object {
                "fontSize": "20px",
                "marginBottom": "7px",
              }
            }
          >
            About us
          </div>
          <div
            className="ant-row"
            style={
              Object {
                "fontSize": "20px",
                "marginBottom": "7px",
              }
            }
          >
            Careers
          </div>
          <div
            className="ant-row"
            style={
              Object {
                "fontSize": "20px",
                "marginBottom": "7px",
              }
            }
          >
            Latest news
          </div>
          <div
            className="ant-row"
            style={
              Object {
                "fontSize": "20px",
                "marginBottom": "7px",
              }
            }
          >
            Partnership
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="ant-divider ant-divider-horizontal"
    role="separator"
  />
  <div
    className="ant-row ant-row-center"
    style={Object {}}
  >
    Copyright 2021. Designed by Frances Yang.
  </div>
</div>
`;
